### ISSUE TITLE
grass: compile with libBLAS

### ISSUE THREAD

**Issue by jascase901** [NixOS/nixpkgs#29081]
## Issue description
some functions related to working with lidar data require blas support
### Steps to reproduce
I am using qgis to run grass functions.

nix-shell -p qgis grass python --run qgis
ctrl-alt-t
search for r.in.lidar
run r.in.lidar with input .las file

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
* Nix version: (run `nix-env --version`)
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`)

**Comment by vbgl** [comment]
Thank you.

Did you test that? It seems that `grass` in `nixpkgs` has been broken for a while. Any hint about how to fix it is welcome.

**Comment by jascase901** [comment]
@vbgl
Sorry I didn't notice this 
It worked on my ubuntu 16.04 machine. I tried again on a centos 7 machine, but it failed complaining about /lib64/libexpat.so. Being impure.

I was able to fix it by hardcoding the location of the nix version of libexpat in its .nix file.

**Comment by jascase901** [comment]
I can link the file tomorrow its not with me right now.

**Comment by jascase901** [comment]
```
{ stdenv, fetchurl, flex, bison, pkgconfig, zlib, libtiff, libpng, fftw
, cairo, readline, ffmpeg, makeWrapper, wxGTK30, netcdf, blas
, proj, gdal, geos, sqlite, postgresql, mysql, python2Packages, libLAS
}:

stdenv.mkDerivation {
  name = "grass-7.0.2";
  src = fetchurl {
    url = http://grass.osgeo.org/grass70/source/grass-7.0.2.tar.gz;
    sha256 = "02qrdgn46gxr60amxwax4b8fkkmhmjxi6qh4yfvpbii6ai6diarf";
  };

  buildInputs = [ flex bison zlib proj gdal libtiff libpng fftw sqlite pkgconfig cairo
  readline ffmpeg makeWrapper wxGTK30 netcdf geos postgresql mysql.client blas libLAS ]
    ++ (with python2Packages; [ python dateutil wxPython30 numpy ]);

  configureFlags = [
    "--with-liblas=${libLAS}/bin/liblas-config"
    "--with-proj-share=${proj}/share/proj"
    "--without-opengl"
    "--with-readline"
    "--with-wxwidgets"
    "--with-netcdf"
    "--with-geos"
    "--with-postgres" "--with-postgres-libs=${postgresql.lib}/lib/"
    # it complains about missing libmysqld but doesn't really seem to need it
    "--with-mysql" "--with-mysql-includes=${stdenv.lib.getDev mysql.client}/include/mysql"
    "--with-blas"
  ];

  /* Ensures that the python script run at build time are actually executable;
   * otherwise, patchShebangs ignores them.  */
  postConfigure = ''
    chmod +x scripts/d.out.file/d.out.file.py \
      scripts/d.to.rast/d.to.rast.py \
      scripts/d.what.rast/d.what.rast.py \
      scripts/d.what.vect/d.what.vect.py \
      scripts/g.extension/g.extension.py \
      scripts/g.extension.all/g.extension.all.py \
      scripts/r.pack/r.pack.py \
      scripts/r.tileset/r.tileset.py \
      scripts/r.unpack/r.unpack.py \
      scripts/v.krige/v.krige.py \
      scripts/v.rast.stats/v.rast.stats.py \
      scripts/v.to.lines/v.to.lines.py \
      scripts/v.what.strds/v.what.strds.py \
      scripts/v.unpack/v.unpack.py \
      scripts/wxpyimgview/*.py \
      gui/wxpython/animation/g.gui.animation.py \
      gui/wxpython/rlisetup/g.gui.rlisetup.py \
      gui/wxpython/vdigit/g.gui.vdigit.py \
      temporal/t.rast.accumulate/t.rast.accumulate.py \
      temporal/t.rast.accdetect/t.rast.accdetect.py \
      temporal/t.select/t.select.py
    for d in gui lib scripts temporal tools; do
      patchShebangs $d
    done
  '';

  postInstall = ''
    wrapProgram $out/bin/grass70 \
    --set PYTHONPATH $PYTHONPATH \
    --set GRASS_PYTHON ${python2Packages.python}/bin/${python2Packages.python.executable} \
    --suffix LD_LIBRARY_PATH ':' '${gdal}/lib'
    ln -s $out/grass-*/lib $out/lib
  '';

  enableParallelBuilding = true;

  meta = {
    homepage = http://grass.osgeo.org/;
    description = "GIS software suite used for geospatial data management and analysis, image processing, graphics and maps production, spatial modeling, and visualization";
    license = stdenv.lib.licenses.gpl2Plus;
    platforms = stdenv.lib.platforms.all;
  };
}
```

**Comment by jascase901** [comment]
I checked it out. the libexpat issue was with a different package. The grass package works on my centos 7 machine as is.

**Comment by luzpaz** [comment]
Can this issue then be closed? (Due to #39027)

**Comment by jascase901** [comment]
Yes
